### PR TITLE
Handle an empty span for 'introduce variable' refactoring

### PR DESCRIPTION
**Pull request by alrz**
Relates to https://github.com/dotnet/roslyn/issues/23310

### CODE REVIEW

**Comment by CyrusNajmabadi** [comment]
1. Add positive tests for things like invoking htis on a ```+``` in a binary expression.
2. Add negative tests that we don't offer to generate local on something like ```var x = y;``` when on the y.  
3. I'm concerned this may walk up too high in some cases.  For exapmle, say i was on the ```=``` in ```() => { var x = y; }```.  You walk up the next expression.  In this case that's the lambda.  That seems wrong to offer this feature for.

File: src/Features/Core/Portable/IntroduceVariable/AbstractIntroduceVariableService.State.cs [hunk 1]
```diff
@@ -186,6 +186,11 @@ public SemanticMap GetSemanticMap(CancellationToken cancellationToken)
             private TExpressionSyntax GetExpressionUnderSpan(SyntaxTree tree, TextSpan textSpan, CancellationToken cancellationToken)
             {
                 var root = tree.GetRoot(cancellationToken);
+                if (textSpan.Length == 0)
+                {
+                    return root.FindToken(textSpan.Start).GetAncestor<TExpressionSyntax>();
```
Comment (CyrusNajmabadi) [inline src/Features/Core/Portable/IntroduceVariable/AbstractIntroduceVariableService.State.cs]: probably better to do "return root.FindToken(...).Parent as TExpressionSyntax;".  That way it works for tokens that are directly parented by expressions (like the => in a lambda), but not some random token (like ```if``` or ```switch```) in a lambda.

**Comment by sharwell** [comment]
@MattGertz for ask mode

**Comment by sharwell** [comment]
@Pilchie for ask mode

**Pull request merged**